main.rb

```diff
@@ -0,0 +1,178 @@
+    # loop through array and pushes elements that are true
+    arr_size.times do |x|
+      new_arr.push(Array(self)[x]) if yield(Array(self)[x])
+    end
+    new_arr
+  end
+
+  # ========
+  #  my_all
+  # ========
+
+  def my_all?(arg = nil)
+    result = false
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result unless i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result unless x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result unless y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result if a.nil? }
+    elsif block_given?
+      my_each { |b| return result unless yield(b) }
+    end
+    !result
+  end
+
+  # ========
+  #  my_any
+  # ========
+
+  def my_any?(arg = nil)
+    result = true
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result if i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result if x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
+
+  # ==========
+  #  my_none?
+  # ==========
+
+  def my_none?(arg = nil)
+    result = false
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result if i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result if x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? || !a }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
+
+  # ==========
+  #  my_count
+  # ==========
+
+  def my_count(count_one = 0)
+    arr_size = Array(self).length
+    count = 0
+
+    unless block_given?
+      return arr_size if count_one.zero?
+
+      arr_size.times do |i|
+        count += 1 if Array(self)[i] == count_one
+      end
+    end
+
+    if block_given? && !Array(self).nil?
+      arr_size.times do |i|
+        count += 1 if yield(Array(self)[i])
+      end
+    end
+    count
+  end
+
+  # ========
+  #  my_map
+  # ========
+
+  def my_map(&_proc)
+    return to_enum unless block_given?
+
+    new_arr = []
+    my_each do |i|
+      new_arr.push(yield(i))
+    end
+    new_arr
+  end
+
+  # ===========
+  #  my_inject
+  # ===========
+
+  def my_inject(accumulator = nil, operator = nil)
+    result = 0
+    arr_size = Array(self).length
+
+    return unless block_given?
+
+    return unless !accumulator.nil? && !operator.is_a?(Symbol) || !operator.is_a?(Integer)
+
+    arr_size.times do |i|
+      return result += yield(sum, Array(self)[i]) if Array(self)[i].class == Integer
+    end
+  end
+end
```

Comment:
- [ ] `my_inject` should **raise** a `localJumpError` when no block or argument is given. Yours is not currently raising any errors.

- [ ] It should also search for the longest word in an array of strings and return that **word**. Yours currently returns an integer.

- [ ] When a block is given without an initial value, `my_inject` should combine all of the elements of enum by applying a binary operation that is specified by a block.
```ruby
[5, 2, 1].inject { |result, x| result + x } == [5, 2, 1].my_inject { |result, x| result + x }  => false
```
- [ ] When `my_inject` is called on a range and a block is given without an initial value, it should combine also all of the elements of enum by applying a binary operation that is specified by a block.

- [ ] When a block is given with an initial value, `my_inject` should combine all of the elements of enum by applying a binary operation that is specified by a block.
```ruby
 [5, 2, 1].inject(0) { |result, x| result + x } ==  [5, 2, 1].my_inject(0) { |result, x| result + x }  => false
```

- [ ] When a symbol is specified without an initial value, `my_inject` should combine each of the elements of the collection by applying the symbol as a named method.

- [ ] When `my_inject` is called on a range, and a symbol is specified without an initial value, it should combine each of the elements of the collection by applying the symbol as a named method.

- [ ] When a symbol is specified with an initial value, `my_inject` should combine each of the elements of the collection by applying the symbol as a named method.

- [ ] When `my_inject` is called on a range, and a symbol is specified with an initial value, it should combine each of the elements of the collection by applying the symbol as a named method.